extensions/amp-analytics/0.1/requests.js

```diff
@@ -133,10 +144,46 @@ export class RequestHandler {
 
     if (!this.baseUrlPromise_) {
       expansionOption.freezeVar('extraUrlParams');
-      this.baseUrlTemplatePromise_ = this.variableService_.expandTemplate(
-        this.baseUrl,
-        expansionOption
-      );
+
+      // expand requestOrigin if it is declared
+      if (this.requestOrigin) {
+        const baseUrlTemplatePromise = this.variableService_.expandTemplate(
+          this.baseUrl,
+          expansionOption
+        );
+
+        // do not encode vars in request origin
+        expansionOption.noEncode = true;
+
+        const requestOriginPromise = this.variableService_
+          // expand variables in request origin
+          .expandTemplate(this.requestOrigin, expansionOption)
+          // substitute in URL values e.g. DOCUMENT_REFERRER -> https://example.com
+          .then(expandedRequestOrigin => {
+            return this.urlReplacementService_.expandUrlAsync(
+              expandedRequestOrigin,
+              bindings,
+              this.whitelist_,
+              true // opt_noEncode
+            );
+          });
+
+        this.baseUrlTemplatePromise_ = Promise.all([
+          requestOriginPromise,
+          baseUrlTemplatePromise,
+        ]).then(promiseValues => {
+          // remove trailing forward slashes in request origin
+          const requestOrigin = promiseValues[0].replace(/\/+$/, '');
+          const baseUrl = promiseValues[1];
+          return requestOrigin + baseUrl;
```

Comment:
We only expand template of the baseUrl here, but have already expanded macro of the requestOrigin. I think it will cleaner if we can add them up after we call `expandUrlAsync` for both. Or before we call `expandUrlAsync` for both.